aptos-move/framework/aptos-stdlib/sources/cryptography/curves.move

```diff
@@ -0,0 +1,217 @@
+module aptos_std::curves {
+    use aptos_std::type_info::type_of;
+
+    // Structs and consts.
+
+    // Fake structs representing group type.
+    struct BLS12_381_G1 {}
+    struct BLS12_381_G2 {}
+    struct BLS12_381_Gt {}
+
+    //TODO: handle as u8 temporarily. Upgrade to u64.
+    struct Scalar<phantom Group> has drop {
+        handle: u8
+    }
+
+    struct Point<phantom Group> has drop {
```

Comment:
I see that there's no `CompressedPoint<T>` struct definition, which would allow us to store a safe-to-deserialize point in storage and potentially optimize the `CompressedPoint<T>` to `Point<T>` conversion instead of just calling `bytes_into_point` which will perform checks to see if the point is on the curve, of prime order, etc.

For now, let's not complicate our lives, since we can always add this feature later.

Reply:
In other words, if devs want to store points, they can simply store the bytes for now. If gas becomes an issue, we can provide a cheaper way to avoid the expensive deserialization (though for some libraries this will not be possible).